llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6208,6 +6212,12 @@ Value *SPIRVToLLVM::transSPIRVImageFetchReadFromInst(SPIRVInstruction *bi, Basic
   auto bii = static_cast<SPIRVImageInstBase *>(bi);
   getImageDesc(bii->getOpValue(0), &imageInfo);
 
+  // OpImageFetch fetches from an image whose Sampled operand is 1. Ops that modify an image (like OpImageWrite) work
```

Comment:
Here, we don't check the Sampled field of the image operand defined by `OpTypeImage`. So the comments don't match the code statements. Actually, for `OpImageFetch`, we don't care the Sampled field because we don't do sampling and the LOD is provided explicitly. HW image_load instruction doesn't need the sampler descriptor. I think we can simplify the comments to say "For OpImageFetch, if the image is known not to alias any other memory object then ...".

Reply:
I know where Jay is coming from with this comment. We were wondering how to identify when certain image ops cannot alias, e.g. is it allowed by the spec that the same `OpTypeImage` is used for both `OpImageFetch` and `OpImageWrite` (it is not allowed as they require images with different `Sampled` value).

Reply:
Thank you for explaining this. Fine to me.

Reply:
Right, I was quoting from the SPIR-V spec to try to justify the argument that OpImageFetch and OpImageWrite can't use the same image. I would be very happy if there was a simpler way of explaining this!